atmat/lattice/atfittune.m

```diff
@@ -22,41 +22,39 @@
 %
 
 [UseIntegerPart,varargin]=getflag(varargin,'UseIntegerPart');
-if isscalar(varargin{1}) && isnumeric(varargin{1})
-    dpp=varargin{1};
-    [newtunes,famname1,famname2]=deal(varargin{2:end});
+[dpp,varargin]=getargs(varargin,[],'check',@(arg) isscalar(arg) && isnumeric(arg));
+[newtunes,famname1,famname2]=deal(varargin{:});
+if isempty(dpp)
+    dpparg={};
 else
-    dpp=0;
-    [newtunes,famname1,famname2]=deal(varargin{:});
+    dpparg={'dp',dpp};
 end
 
 idx1=varelem(ring,famname1);
 idx2=varelem(ring,famname2);
 
 kl1=atgetfieldvalues(ring(idx1),'PolynomB',{2});
 kl2=atgetfieldvalues(ring(idx2),'PolynomB',{2});
-delta = 1e-6;
-if ~UseIntegerPart
+delta = 1e-6;   % Step on quadrupole strengths
```

Comment:
This could be an optional input defaulted to this value?

Reply:
Done